zkevm-circuits/src/evm_circuit.rs

```diff
@@ -510,6 +527,27 @@ impl<F: Field> Circuit<F> for EvmCircuit<F> {
             block.circuits_params.max_rws,
             challenges.evm_word(),
         )?;
+
+        #[cfg(feature = "dual_bytecode")]
+        {
+            // when enable feature "dual_bytecode", get two sets of bytecodes here.
+            let (first_bytecodes, second_bytecodes) = block.get_bytecodes_for_dual_sub_circuits();
+            if !first_bytecodes.is_empty() {
+                // assign first bytecode_table
+                config
+                    .bytecode_table
+                    .dev_load(&mut layouter, first_bytecodes, &challenges)?;
+            }
+
+            if !second_bytecodes.is_empty() {
```

Comment:
Just a note (to double check):

- Do we use `dev_load` from bytecode table even for production?
- If there are no bytecodes, do we _simply not_ call `dev_load`? Or do we call it with empty slice?

In your case, you are _not_ calling it if bytecodes are not present. May be we need to call using an empty slice, i.e. if there is any kind of header/placeholder rows that are being populated?

You can resolve this comment if its irrelevant, but better to double check

Reply:
1. dev_load just use for unit testing, not for  super circuit production 
2. currently no such case:  care about header/placeholder for empty bytecode, but anyway can always call with empty slice.